XiEditor/EditView.swift

```diff
@@ -248,6 +248,15 @@ class EditView: NSView, NSTextInputClient {
                 }
             }
         }
+
+        // Update cursorPos
+        // TODO: How should we handle the situations that have multi-cursor?
+        for lineIx in first..<last {
+            guard let line = getLine(lineIx) else { continue }
```

Comment:
Thanks for this catch!

Two little things: instead of running the loop again, you can set `cursorPos` when we draw the cursor(s), around line 205 in master. That will also let you use the utf-16 offset that is calculated there (the `cix` variable), to make sure the IME window lines up correctly with the text. (Try this out and see if it isn't more correct, the window should line up better with the input.)